src/Orchard.ContentManagement/DefaultContentManager.cs

```diff
@@ -420,6 +428,14 @@ public Task DiscardDraftAsync(ContentItem contentItem)
 
             Handlers.Reverse().Invoke(handler => handler.Removed(context), _logger);
 
+            var publishedItem = GetAsync(contentItem.ContentItemId, VersionOptions.Published).GetAwaiter().GetResult();
```

Comment:
Yes but `DiscardDraftAsync()` is not decalred as `async` and return `Task.CompletedTask`.
I thought there was some reasons, otherwise i can update this and retry it.